### ISSUE TITLE
Lighthouse panic: runtime error: invalid memory address or nil pointer dereference [terraform-provider-azurerm_v3.102.0_x5]

### ISSUE THREAD

**Issue by alexeyovriakh** [hashicorp/terraform-provider-azurerm#25998]
### Is there an existing issue for this?

- [X] I have searched the existing issues

### Community Note

* Please vote on this issue by adding a :thumbsup: [reaction](https://blog.github.com/2016-03-10-add-reactions-to-pull-requests-issues-and-comments/) to the original issue to help the community and maintainers prioritize this request
* Please do not leave "+1" or "me too" comments, they generate extra noise for issue followers and do not help prioritize the request
* If you are interested in working on this issue or have submitted a pull request, please leave a comment and review the [contribution guide](https://github.com/hashicorp/terraform-provider-azurerm/blob/main/contributing/README.md) to help.

```
resource "azurerm_lighthouse_definition" "lighthouse_definition" {
  name               = "<...>"
  description        = "<...>"
  managing_tenant_id = var.id
  scope              = "/subscriptions/${var.subscription_id}"
  authorization {
    principal_id           = var.service_principal_id
    principal_display_name = "<...>"
    # Uses VM Scanner Operator role
    role_definition_id = "d24ecba3-c1f4-40fa-a7bb-4588a071e8fd"
  }
}

resource "azurerm_lighthouse_assignment" "lighthouse_assignment" {
  count = var.is_organizational ? 0 : 1
  scope                    = "/subscriptions/${var.subscription_id}"
  lighthouse_definition_id = azurerm_lighthouse_definition.lighthouse_definition.id
}
```

### Terraform Version

>= 1.0.0

### AzureRM Provider Version

3.102.0

### Affected Resource(s)/Data Source(s)

azurerm_lighthouse_definition

### Terraform Configuration Files

```hcl
-
```

### Debug Output/Panic Output

```shell
Stack trace from the terraform-provider-azurerm_v3.102.0_x5 plugin:

panic: runtime error: invalid memory address or nil pointer dereference
[signal SIGSEGV: segmentation violation code=0x2 addr=0x8 pc=0x1099aacdc]

goroutine 35 [running]:
github.com/hashicorp/terraform-provider-azurerm/internal/services/lighthouse.resourceLighthouseDefinitionCreateUpdate(0x0?, {0x10b3ef020?, 0x14002842480?})
	github.com/hashicorp/terraform-provider-azurerm/internal/services/lighthouse/lighthouse_definition_resource.go:221 +0x5c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).create(0x10c40f4f8?, {0x10c40f4f8?, 0x14002486cc0?}, 0xd?, {0x10b3ef020?, 0x14002842480?})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:766 +0x134
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*Resource).Apply(0x140015f5180, {0x10c40f4f8, 0x14002486cc0}, 0x140027b2f70, 0x140017b8780, {0x10b3ef020, 0x14002842480})
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/resource.go:909 +0x86c
github.com/hashicorp/terraform-plugin-sdk/v2/helper/schema.(*GRPCProviderServer).ApplyResourceChange(0x140004da0f0, {0x10c40f4f8?, 0x14002486bd0?}, 0x14001424000)
	github.com/hashicorp/terraform-plugin-sdk/v2@v2.29.0/helper/schema/grpc_provider.go:1060 +0xb08
github.com/hashicorp/terraform-plugin-go/tfprotov5/tf5server.(*server).ApplyResourceChange(0x14001ac5220, {0x10c40f4f8?, 0x140024861b0?}, 0x14002754070)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/tf5server/server.go:859 +0x3c0
github.com/hashicorp/terraform-plugin-go/tfprotov5/internal/tfplugin5._Provider_ApplyResourceChange_Handler({0x10c02f340?, 0x14001ac5220}, {0x10c40f4f8, 0x140024861b0}, 0x14002754000, 0x0)
	github.com/hashicorp/terraform-plugin-go@v0.19.0/tfprotov5/internal/tfplugin5/tfplugin5_grpc.pb.go:467 +0x164
google.golang.org/grpc.(*Server).processUnaryRPC(0x1400047c000, {0x10c438280, 0x140018601a0}, 0x1400248a000, 0x14001791440, 0x111b7f098, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1374 +0xb44
google.golang.org/grpc.(*Server).handleStream(0x1400047c000, {0x10c438280, 0x140018601a0}, 0x1400248a000, 0x0)
	google.golang.org/grpc@v1.58.3/server.go:1751 +0x7cc
google.golang.org/grpc.(*Server).serveStreams.func1.1()
	google.golang.org/grpc@v1.58.3/server.go:986 +0xb4
created by google.golang.org/grpc.(*Server).serveStreams.func1 in goroutine 52
	google.golang.org/grpc@v1.58.3/server.go:997 +0x160

Error: The terraform-provider-azurerm_v3.102.0_x5 plugin crashed!

This is always indicative of a bug within the plugin. It would be immensely
helpful if you could report the crash with the plugin's maintainers so that it
can be fixed. The output above should help diagnose the issue.
```

**Comment by tombuildsstuff** [comment]
hey @alexeyovriakh 

A fix for this crash has been shipped and is available in a newer version of the Provider (`v3.103.1` being the latest at the time of writing) - would you mind updating to the latest version of the Provider and see if this fixes this for you?

Since a fix for this has shipped and I believe this'll be fixed by updating to use the latest version of the Provider, I'm going to close this issue for the moment - but if that doesn't fix this for you then please let us know and we can take another look.

Thanks!

**Comment by alexeyovriakh** [comment]
Thanks so much, @tombuildsstuff!
